src/software/new_geom/util/intersects_test.cpp

```diff
@@ -53,12 +53,15 @@ TEST(IntersectsTest, segment_overlapping_polygon_edge)
     EXPECT_TRUE(intersects(s, p));
 }
 
-TEST(IntersectsTest, segment_contained_in_polygon)
+// This tests undefined behaviour so it is DISABLED_, but can be useful for debugging
```

Comment:
Could you detail a little more about what exactly about this is undefined, and why it's helpful enough to warrant the dead code?

Reply:
It's undefined, because the interface for intersects now allows true or false. Similar to the invariantsHold, these are good for debugging, but I wouldn't mind removing them either

Reply:
> the interface for intersects now allows true or false

don't understand what you mean by this?

Reply:
See the new definitions: https://github.com/UBC-Thunderbots/Software/pull/1443/files#diff-3e0d25e5e5bcf5bf568428e6fb809b02

Reply:
What's the performance loss if we check if the point is within the polygon? I would much rather we have a `edgesIntersect` (or something) function that performs this check, and then have the generic `intersects` check `edgesIntersect` and the end points (so we'd still end up with a separate utility function, just changing the default functionality of `intersects`)

Reply:
3X slowdown on polygons and 2X slowdown on circles

Reply:
Can't easily rename it because of how obstacle is implemented. I also don't think it's necessary, because none of the users of intersects mind (tests pass), and if they do care, they can do everything they wanted to previously by explicitly checking with contains.

Reply:
> See the new definitions: #1443 (files) If a Segment is fully within a shape, then intersects makes no guarantees.

Ah I see thanks for clarifying. I agree (to what I think @garethellis0 is saying) that I'd prefer to leave the interface of the existing `intersects` the same so it's consistent with the rest of them in `geom`.

Reply:
Let's discuss this offline

Reply:
Offline consensus was that Ray-Segment intersection seems to be what's slowing down the `contains` check for polygons, so @jonathanlew is going to look at making it faster, and then if it's still 2x slower or worse we'll chat about other options.

Reply:
@garethellis0 @MathewMacDougall I've added back the contains check and it has a 20% performance hit so we're probably slightly slower than before this PR, but that should be ok